aana/deployments/task_queue_deployment.py

```diff
@@ -61,81 +58,82 @@ async def apply_config(self, config: dict[str, Any]):
         The configuration should conform to the TaskQueueConfig schema.
         """
         config_obj = TaskQueueConfig(**config)
-        self.handle = None
         self.app_name = config_obj.app_name
-        self.thread_pool = concurrent.futures.ThreadPoolExecutor(
-            max_workers=aana_settings.task_queue.num_workers,
-        )
 
     async def loop(self):  # noqa: C901
         """The main loop for the task queue deployment.
 
         The loop will check the queue and assign tasks to workers.
         """
-
-        async def handle_task(task_id: str):
-            """Process a task."""
-            # Fetch the task details
-            task = self.task_repo.read(task_id)
-            # Initially set the task status to RUNNING
-            self.task_repo.update_status(task_id, TaskStatus.RUNNING, 0)
-            try:
-                # Call the endpoint asynchronously
-                out = await self.handle.call_endpoint.remote(task.endpoint, **task.data)
-                # Update the task status to COMPLETED
-                self.task_repo.update_status(task_id, TaskStatus.COMPLETED, 100, out)
-            except Exception as e:
-                # Handle the exception and update the task status to FAILED
-                error_response = custom_exception_handler(None, e)
-                error = orjson.loads(error_response.body)
-                self.task_repo.update_status(task_id, TaskStatus.FAILED, 0, error)
-
-        def run_handle_task(task_id):
-            """Wrapper to run the handle_task function."""
-            run_async(handle_task(task_id))
-
-        def is_thread_pool_full():
-            """Check if the thread pool has too many tasks.
-
-            We use it to stop assigning tasks to the thread pool if it's full
-            to prevent the thread pool from being overwhelmed.
-            We don't want to schedule all tasks from the task queue (could be millions).
-            """
-            return (
-                self.thread_pool._work_queue.qsize()
-                > aana_settings.task_queue.num_workers * 2
-            )
+        running_task_ids: list[str] = []
+        handle = None
+
+        active_tasks = self.task_repo.get_active_tasks()
+        for task in active_tasks:
+            if task.status == TaskStatus.RUNNING:
+                running_task_ids.append(task.id)
+            if task.status == TaskStatus.ASSIGNED:
+                self.task_repo.update_status(
+                    task_id=task.id,
+                    status=TaskStatus.NOT_FINISHED,
+                    progress=0,
+                )
 
         while True:
             if not self._configured:
                 # Wait for the deployment to be configured.
                 await asyncio.sleep(1)
                 continue
 
-            # Remove completed tasks from the futures dictionary
-            for task_id in list(self.futures.keys()):
```

Comment:
Yuo removed this part but keep the logic in the __del__ function

Reply:
Replaced with deployment response cancellation.